de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatWorkspaceCommandStack.java

```diff
@@ -224,20 +227,22 @@ public void setEditingDomain(VirSatTransactionalEditingDomain editingDomain) {
 	/**
 	 * Trigger a save all upon completion of the executing command
 	 */
-	public void triggerSaveAll() {
+	public synchronized void triggerSaveAll() {
+		Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "VirSatWorkspaceCommandStack: Save all Triggered by Thread: " + Thread.currentThread().getName()));
 		triggerSave = true;
 	}
 	
 	/**
 	 * Check if the triggerSave flag has been set. If so, ask the editing domain to save everything
 	 * and reset the flag.
 	 */
-	private void checkTriggerSaveAll() {
+	private synchronized void checkTriggerSaveAll() {
+		Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "VirSatWorkspaceCommandStack: Check for save trigger"));
 		if (triggerSave) {
-			Activator.getDefault().getLog().log(new Status(IStatus.INFO, Activator.getPluginId(), "Executed command triggered save all on editing domain"));
-			VirSatSaveJob saveJob = new VirSatSaveJob(editingDomain, true);
-			saveJob.scheduleIfNoSaveJobPending();
-			triggerSave = false;
+			Activator.getDefault().getLog().log(new Status(IStatus.INFO, Activator.getPluginId(), "VirSatWorkspaceCommandStack: Save tiggered Scheduling Save Job."));
+			// This boolean has to be reset first, otherwise the save all will be executed in

```

Comment:
Which boolean are we talking about in this comment?

Reply:
Has been a deprecated comment. Updated it.

Reply:
@PhilMFischer I dont see any new commits.